projects/packages/jetpack-mu-wpcom/changelog/add-cache-domains-list-request-response

```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: added
+
+Cache the response of the Domain List request"
```

Comment:
Nit: we should remove the trailing `"`, and reflect the other hardening changes to make the code more robust
```suggestion
Cache the response of the Domain List request, and harden the code
```